arcgis-ios-sdk-samples/Maps/Download preplanned map/DownloadPreplannedMapViewController.swift

```diff
@@ -0,0 +1,243 @@
+    static let popoverPreferredRegularHeight: CGFloat = 360
+}
+
+// MARK: - DownloadPreplannedMapViewController
+
+class DownloadPreplannedMapViewController: UIViewController {
+    @IBOutlet private weak var mapView: AGSMapView!
+    @IBOutlet private weak var activityIndicatorView: UIActivityIndicatorView!
+    @IBOutlet private weak var selectMapBarButtonItem: UIBarButtonItem!
+    @IBOutlet private weak var removeDownloadsBarButtonItem: UIBarButtonItem!
+
+    private let portal = AGSPortal.arcGISOnline(withLoginRequired: false)
+    private var portalItem: AGSPortalItem?
+    private var onlineMap: AGSMap?
+
+    private var offlineTask: AGSOfflineMapTask?
+    private var cancelable: AGSCancelable?
+
+    private var remoteAvailablePreplannedMapAreas = [AGSPreplannedMapArea]()
+
+    private var remoteLoadedPreplannedMapAreas = [AGSPreplannedMapArea]() {
+        didSet {
+            updateView()
+        }
+    }
+
+    private var uniqueLocalMapPackages = Set<AGSMobileMapPackage>() {
+        didSet {
+            updateView()
+        }
+    }
+
+    // MARK: UIViewController
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if let navigationController = segue.destination as? UINavigationController,
+            let viewController = navigationController.viewControllers.first as? MapSelectionTableViewController {
+            navigationController.presentationController?.delegate = self
+
+            let height: CGFloat
+            if traitCollection.horizontalSizeClass == .regular, traitCollection.verticalSizeClass == .regular {
+                height = Constants.popoverPreferredRegularHeight
+            } else {
+                height = Constants.popoverPreferredCompactHeight
+            }
+            let contentSize = CGSize(width: Constants.popoverPreferredWidth, height: height)
+            viewController.preferredContentSize = contentSize
+
+            viewController.mapSelectionDelegate = self
+            viewController.currentlySelectedMap = mapView.map
+            viewController.onlineMap = onlineMap
+            viewController.availablePreplannedMapAreas = remoteLoadedPreplannedMapAreas
+            viewController.localMapPackages = Array(uniqueLocalMapPackages)
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        guard let sourceBarButtonItem = navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem else {
+            return
+        }
+        sourceBarButtonItem.filenames = ["DownloadPreplannedMapViewController", "MapSelectionTableViewController", "PreplannedMapAreaTableViewCell"]
+
+        activityIndicatorView.color = Constants.activityIndicatorColor
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        loadPortal()
+    }
+
+    // MARK: IBActions
+
+    @IBAction private func removeDownloadsTapped(_ sender: UIBarButtonItem) {
```

Comment:
App crash on clicking `Delete Maps` button

```
* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGABRT
    frame #0: 0x0000000215dec0cc libsystem_kernel.dylib`__pthread_kill + 8
    frame #1: 0x0000000215e62a88 libsystem_pthread.dylib`pthread_kill + 300
    frame #2: 0x0000000215d4514c libsystem_c.dylib`abort + 144
    frame #3: 0x0000000215446f78 libc++abi.dylib`abort_message + 136
    frame #4: 0x000000021544713c libc++abi.dylib`default_terminate_handler() + 320
    frame #5: 0x000000021545f144 libobjc.A.dylib`_objc_terminate() + 124
    frame #6: 0x00000002154539cc libc++abi.dylib`std::__terminate(void (*)()) + 20
    frame #7: 0x00000002154535c4 libc++abi.dylib`__cxa_rethrow + 148
    frame #8: 0x000000021545eee0 libobjc.A.dylib`objc_exception_rethrow + 48
    frame #9: 0x00000002161e32c0 CoreFoundation`CFRunLoopRunSpecific + 560
    frame #10: 0x0000000218422d8c GraphicsServices`GSEventRunModal + 108
    frame #11: 0x00000002432134c0 UIKitCore`UIApplicationMain + 216
  * frame #12: 0x0000000102f58c84 ArcGIS Runtime SDK Samples`main at AppDelegate.swift:19:7
    frame #13: 0x0000000215c9ffd8 libdyld.dylib`start + 4
```

Reply:
@saratchandrakarumuri I'm sorry that you encountered a crash. 

I'm unable to reproduce this at the moment; can you please share some additional detail? For example, Xcode & iOS versions, device versus Simulator, etc.

Reply:
@philium did you encounter a crash during your review?

Reply:
This is the setup I currently have and I am still able to get the crash

Xcode - 10.2
iOS - 12.3.1
Device - iPad Pro 11"

Reply:
Thanks @saratchandrakarumuri. By any chance, was the popover still visible when you tapped that button (i.e., the "trash can")?

Reply:
Nope. That button is disabled when the popover is open and only after the dismissal of popover, the button is enabled

Reply:
Gotcha - thank you for the follow up. While I have not been able to reproduce the issue myself, I have pushed two commits that should have a beneficial effect here : both [39bc11df](https://github.com/Esri/arcgis-runtime-samples-ios/pull/758/commits/39bc11df40383ac8276e8d56fd4deae36b037431) & [f02b24a3](https://github.com/Esri/arcgis-runtime-samples-ios/pull/758/commits/f02b24a3882749039341532e25b13ec7e0508b4e).